[MA-913] update slo doc (#6818)
DataDog/documentation: content/en/monitors/service_level_objectives/monitor.md

Here we add contextual information about the purpose of the SLO, including any related information
in the description and tags you would like to associate with the SLO.

#### Underlying monitor and SLO histories

Making changes to the monitor used by an SLO recalculates the SLO history.  Therefore, the monitor history and SLO history may not match after a monitor update.

> SLO calculations do not take into account when a monitor is resolved manually or as a result of the **_After x hours automatically resolve this monitor from a triggered state_** setting.  If these are important tools for your workflow, consider cloning your monitor, removing auto-resolve settings and `@-notification`s, and using the clone for your SLO.

Reviewer: Or should we suggest using metric-based SLOs?

Reviewer 2: We shouldn't suggest it because it depends on the use case but we can remind them to determine if they are using the right one. 

Confirm you are using the preferred SLI type for your use case. In Datadog we support monitor-based SLIs as described here or you can also use metric-based SLIs as described here(https://docs.datadoghq.com/monitors/service_level_objectives/event/)

Author: Added @meghanelizabeth 's suggestion below

Here we add contextual information about the purpose of the SLO, including any related information
in the description and tags you would like to associate with the SLO.

#### Underlying monitor and SLO histories

Making changes to the monitor used by an SLO recalculates the SLO history.  Therefore, the monitor history and SLO history may not match after a monitor update.

We recommend against using monitors with `Alert Recovery Threshold` and `Warning Recovery Threshold` as they can also affect your SLO calculations and do not allow you to cleanly differentiate between a SLI's good behavior and bad behavior.

SLO calculations do not take into account when a monitor is resolved manually or as a result of the **_After x hours automatically resolve this monitor from a triggered state_** setting.  If these are important tools for your workflow, consider cloning your monitor, removing auto-resolve settings and `@-notification`s, and using the clone for your SLO.

> Confirm you are using the preferred SLI type for your use case. In Datadog we support monitor-based SLIs as described here or you can also use metric-based SLIs as described [here][5].

Author: @cswatt I added below.  Maybe I didn't use the right syntax?

> Confirm you are using the preferred SLI type for your use case. Datadog supports monitor-based SLIs as described here. You can also use metric-based SLIs as [described in the SLO event documentation][5].

Reviewer 3: `Datadog supports monitor-based SLIs as described here`

To which page the here should point to? is it the same as the `[5]` reference?

Reviewer 3: ping @unclebconnor

Author: @l0k0ms  Yep that's the one.

Reviewer 3: ```suggestion
Confirm you are using the preferred SLI type for your use case. Datadog supports monitor-based SLIs and metric-based SLIs as [described in the SLO event documentation][5].
```
